packages/flutter/lib/src/widgets/spell_check.dart

```diff
@@ -201,20 +211,106 @@ TextSpan buildTextSpanWithSpellCheckSuggestions(
         value.text, spellCheckResultsText, spellCheckResultsSpans);
   }
 
-  return TextSpan(
+  // We will draw the TextSpan tree based on the composing region, if it is
+  // available.
+  final bool shouldConsiderComposingRegion = defaultTargetPlatform == TargetPlatform.android;
```

Comment:
I asked about `TargetPlatform.android` including Samsung because I don't understand why this would crash? Does the Samsung keyboard not have a composing region? If so, I didn't realize that.

I still need to test this, but it should not be a problem because in addition to this check, I can just check if the composing region is valid before using the method that considers it.

Reply:
It does have a composing region, it just doesn't use it for the current editing word (as far as I know).  Maybe I'm too pessimistic.  I thought the Samsung keyboard wasn't compatible at the engine level or something.  If it works then that's great.